src/main/kotlin/template/AnkDokkaPlugin.kt

```diff
@@ -0,0 +1,127 @@
+package template
+
+//import com.github.nomisrev.engine.Engine
+//import com.github.nomisrev.engine.Snippet
+//import com.github.nomisrev.engine.fenceRegexStart
+import java.net.URLClassLoader
+import javax.script.ScriptEngineManager
+import org.jetbrains.dokka.base.DokkaBase
+import org.jetbrains.dokka.model.DModule
+import org.jetbrains.dokka.model.doc.Br
+import org.jetbrains.dokka.model.doc.CodeBlock
+import org.jetbrains.dokka.model.doc.Text
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.DokkaPlugin
+import org.jetbrains.dokka.transformers.documentation.PreMergeDocumentableTransformer
+
+class AnkDokkaPlugin : DokkaPlugin() {
+    val dokkaBasePlugin by lazy { plugin<DokkaBase>() }
+    val ank by extending {
+        dokkaBasePlugin.preMergeDocumentableTransformer providing ::AnkCompiler
+    }
+
+    init { // This is optional and experimental, but may save some memory.
+        System.setProperty("kotlin.jsr223.experimental.resolve.dependencies.from.context.classloader", "true")
+    }
+}
+
+class AnkCompiler(val ctx: DokkaContext) : PreMergeDocumentableTransformer {
+
+    // Could we optimise with `suspend` and running in parallel?
+    override fun invoke(modules: List<DModule>): List<DModule> =
+        modules.also {
+            val classLoader = classLoader()
+
+            // We need to get the original contextClassLoader which Dokka uses to run, and store it
+            // Then we need to set the contextClassloader so the engine can correctly define the compilation classpath
+            // We do this **whilst** decoupling the classLoader for the ScriptEngine with the classLoader of Dokka.
+            // This is because Dokka shadows some of the Kotlin compiler dependencies, having them mixed results in incorrect state
+            val originalClassLoader = Thread.currentThread().contextClassLoader
+            Thread.currentThread().contextClassLoader = classLoader
+
+            ScriptEngineManager(classLoader)
+                .getEngineByExtension("kts")
+                .eval("println(\"HELLO\")")
+
+            // When we're done, we reset back to the original classLoader
```

Comment:
I'd recommend try/finally here, the eval may still throw an exception

Reply:
Thanks, @ligee, I've cleaned the code up a bit and going to merge.

Here is the "engine" bit, https://github.com/nomisRev/AnkDokkaPlugin/pull/1/files#diff-0da536be0ad16caf98f40e9b19f6836d616941842b2e12cb14561b7190d9374fR37.

I wrapped this code in `try/finally, thanks for the tip. I'm also able to import all dependencies from the users classpath into the ScriptEngine without any problems.

As a next experiment, I'm going to attempt to refactor the js223 module and try to load the jar directly onto the classpath.